lib/daru/vector.rb

```diff
@@ -964,7 +964,7 @@ def reindex! new_index
     #   #   c   3
     def reorder! order
       @index = @index.reorder order
-      @data = order.map { |i| @data[i] }
+      @data = Accessors::ArrayWrapper.new(order.map { |i| @data[i] }, self)
```

Comment:
Good catch.

The type of `@data` should depend on the value of the of the `@dtype` variable. Could you just add this line instead of your change:
``` ruby
cast_vector_to @dtype, @data, @nm_dtype
```